Header fixes

NaominoDino opened the pull request:
I fixed some one the build errors. I’ll keep going on it!

NaominoDino committed:
Update symbols_resolver.h

Missing Default Return Type. Fixed

NaominoDino committed:
Update libkernel.cpp

Changing from char* to const char*
Regarding C++ ISO

NaominoDino committed:
Update condition.h

Missing default Return statement

NaominoDino committed:
Update liverpool_to_vk.cpp

Missing Default Statement for Stability
Exception throw (Later draw it to TextBox instead of exception -> Prevent crash)

raphaelthegreat commented on src/video_core/renderer_vulkan/liverpool_to_vk.cpp:
```diff
@@ -222,6 +222,9 @@ vk::CompareOp DepthCompare(AmdGpu::DepthCompare comp) {
         return vk::CompareOp::eAlways;
+    /// Missing AMG Gpu default statement
+    default:
+        throw std::invalid_argument("Unsupported DeepCompare");
```
```suggestion
    default:
        UNREACHABLE_MSG("Unknown depth compare op {}", u32(comp));
```

raphaelthegreat commented on src/core/libraries/kernel/libkernel.cpp:
```diff
@@ -295,8 +295,9 @@ int PS4_SYSV_ABI sceKernelUuidCreate(OrbisKernelUuid* orbisUuid) {
 
-char* PS4_SYSV_ABI sceKernelGetFsSandboxRandomWord() {
-    char* path = "sys";
+/// Naomi Rework (Const)
```
```suggestion
```

NaominoDino committed:
Update src/video_core/renderer_vulkan/liverpool_to_vk.cpp

Good Change, thanks for suggestion

Co-authored-by: TheTurtle <47210458+raphaelthegreat@users.noreply.github.com>

NaominoDino committed:
Update liverpool_to_vk.cpp

Update

NaominoDino committed:
Merge branch 'main' of https://github.com/Naomi-Kali/shadPS4

NaominoDino committed:
Thread_management rework

Only error Left:
/home/naomi/Asztal/shadPS4/GitHub/shadPS4/src/core/libraries/kernel/thread_management.cpp: In function ‘int Libraries::Kernel::scePthreadAttrGetstackaddr(PthreadAttrInternal* const*, void**)’:
/home/naomi/Asztal/shadPS4/GitHub/shadPS4/src/core/libraries/kernel/thread_management.cpp:321:43: warning: ‘int pthread_attr_getstackaddr(const pthread_attr_t*, void**)’ is deprecated [-Wdeprecated-declarations]
  321 |     int result = pthread_attr_getstackaddr(&(*attr)->pth_attr, stack_addr);
      |                  ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/14.1.1/x86_64-pc-linux-gnu/bits/gthr-default.h:35,
                 from /usr/include/c++/14.1.1/x86_64-pc-linux-gnu/bits/gthr.h:157,
                 from /usr/include/c++/14.1.1/bits/std_mutex.h:41,
                 from /usr/include/c++/14.1.1/mutex:47,
                 from /home/naomi/Asztal/shadPS4/GitHub/shadPS4/src/core/libraries/kernel/thread_management.cpp:4:
/usr/include/pthread.h:353:12: note: declared here
  353 | extern int pthread_attr_getstackaddr (const pthread_attr_t *__restrict

/usr/bin/ld: CMakeFiles/shadps4.dir/src/core/libraries/kernel/thread_management.cpp.o: in function `Libraries::Kernel::scePthreadAttrGetstackaddr(Libraries::Kernel::PthreadAttrInternal* const*, void**)':
thread_management.cpp:(.text+0x38b): warning: the use of `pthread_attr_getstackaddr' is deprecated, use `pthread_attr_getstack'

raphaelthegreat commented on src/core/libraries/kernel/thread_management.cpp:
```diff
@@ -343,6 +345,18 @@ int PS4_SYSV_ABI scePthreadAttrSetstackaddr(ScePthreadAttr* attr, void* addr) {
 
     return result == 0 ? SCE_OK : SCE_KERNEL_ERROR_EINVAL;
+}*/
+int PS4_SYSV_ABI scePthreadAttrSetstackaddr(ScePthreadAttr* attr, void* stack_addr, size_t stack_size) {
```
This change is wrong, please revert it

NaominoDino replied:
Done, reverted

NaominoDino committed:
Revert "Thread_management rework"

This reverts commit a5b3967fd2dbb0b280d2e11b4ced0acc1269d3df.

NaominoDino committed:
My Main push

Just my Main Push

NaominoDino committed:
Merge branch 'main' of https://github.com/Naomi-Kali/shadPS4

NaominoDino committed:
Hide Cursor

Hiding Cursor implemented

NaominoDino committed:
HideCursor Config

Option inside the config to hide/show cursor during SDL3

raphaelthegreat commented on src/sdl_window.cpp:
```diff
@@ -11,9 +11,18 @@
 
 namespace Frontend {
 
+void WindowSDL::CursorVisibility(bool cursor_visibility) {
```
```suggestion
void WindowSDL::SetCursorVisibility(bool cursor_visibility) {
```

NaominoDino committed:
Update sdl_window.cpp

Co-authored-by: TheTurtle <47210458+raphaelthegreat@users.noreply.github.com>
